wdl-analysis/src/handlers/goto_definition.rs

```diff
@@ -0,0 +1,660 @@
+        let (target_expr, member_ident) = access_expr.operands();
+
+        if member_ident.span() != token.span() {
+            return Ok(None);
+        }
+
+        let scope = analysis_doc
+            .find_scope_by_position(parent_node.span().start())
+            .ok_or_else(|| anyhow!("could not find scope for access expression"))?;
+
+        let mut ctx = GotoDefEvalContext {
+            scope,
+            document: analysis_doc,
+        };
+        let mut evaluator = ExprTypeEvaluator::new(&mut ctx);
+        let target_type = evaluator
+            .evaluate_expr(&target_expr)
+            .unwrap_or(crate::types::Type::Union);
+
+        if let Some(struct_ty) = target_type.as_struct() {
+            let struct_def = analysis_doc
+                .structs()
+                .find(|(_, s)| {
+                    if let Some(s_ty) = s.ty() {
+                        if let Some(s_struct_ty) = s_ty.as_struct() {
+                            return s_struct_ty.name() == struct_ty.name();
+                        }
+                    }
+                    s.name() == struct_ty.name().as_str()
+                })
+                .map(|(_, s)| s)
+                .ok_or_else(|| anyhow!("struct definition not found for {}", struct_ty.name()))?;
+
+            let (uri, def_lines) = match struct_def.namespace() {
+                Some(ns_name) => {
+                    // SAFETY: `namepsace` returns `Some` only when struct was imported from a
+                    // namepsace that exists in the document.
+                    let ns = analysis_doc.namespace(ns_name).unwrap();
+
+                    // SAFETY: `ns.source` comes from a valid namepsace entry which gurantees the
+                    // document exists in the graph.
+                    let imported_node = graph.get(graph.get_index(ns.source()).unwrap());
+
+                    // SAFETY: we successfully got the document above, it's in
+                    // `ParseState::Parsed` which always has a valid lines field.
+                    let lines = imported_node.parse_state().lines().unwrap().clone();
+                    (ns.source().as_ref(), lines)
+                }
+                None => (document_uri, lines.clone()),
+            };
+
+            let struct_node =
+                v1::StructDefinition::cast(SyntaxNode::new_root(struct_def.node().clone()))
+                    .expect("should cast to struct definition");
+
+            if let Some(member) = struct_node
+                .members()
+                .find(|m| m.name().text() == member_ident.text())
+            {
+                let member_span = member.name().span();
+                let span = Span::new(member_span.start() + struct_def.offset(), member_span.len());
+                // Returns found struct member definition location.
+                return Ok(Some(location_from_span(uri, span, &def_lines)?));
+            }
```

Comment:
Nit (so that we don't fall through):

```suggestion
           let Some(member) = struct_node
                .members()
                .find(|m| m.name().text() == member_ident.text()) else {
                return Ok(None);
            };

            let member_span = member.name().span();
            let span = Span::new(member_span.start() + struct_def.offset(), member_span.len());
            // Returns found struct member definition location.
            return Ok(Some(location_from_span(uri, span, &def_lines)?));
        }
```